src/JsonSchema/Constraints/ObjectConstraint.php

```diff
@@ -125,7 +125,7 @@ public function validateElement($element, $matches, $objectDefinition = null, $p
     public function validateDefinition($element, $objectDefinition = null, $path = null)
     {
         foreach ($objectDefinition as $i => $value) {
-            $property = $this->getProperty($element, $i, new UndefinedConstraint());
+            $property = $this->getProperty($element, $i, $this->getFactory()->createInstanceFor('undefined'));
```

Comment:
It's a necessary change, the factory passes existing schemaStorage to the new constaint